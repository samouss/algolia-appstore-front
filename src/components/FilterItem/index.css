.FilterItem {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  padding: 5px 10px;

  &:hover,
  &:hover span {
    color: #0080ff;
  }
}

.FilterItem--active,
.FilterItem--active span {
  background-color: #0080ff;
  color: #ffffff;

  &:hover,
  &:hover span {
    color: #ffffff;
  }
}
