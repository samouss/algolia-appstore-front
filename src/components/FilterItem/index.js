import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './index.css';

const FilterItem = ({
  isRefined,
  children,
  onChange,
}) => {
  const onClick = event => {
    event.preventDefault();
    onChange();
  };

  return (
    <li>
      <a
        href=""
        styleName={cx('FilterItem', {
          'FilterItem--active': isRefined,
        })}
        onClick={onClick}
      >
        {children}
      </a>
    </li>
  );
};

FilterItem.propTypes = {
  isRefined: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default FilterItem;
