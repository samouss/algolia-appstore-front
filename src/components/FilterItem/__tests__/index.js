import React from 'react';
import { shallow } from 'enzyme';
import FilterItem from '../index';

describe('<FilterItem />', () => {
  const defaultProps = {
    isRefined: false,
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <FilterItem
        {...props}
      >
        Value
      </FilterItem>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render when refined', () => {
    const props = {
      ...defaultProps,
      isRefined: true,
    };

    const component = shallow(
      <FilterItem
        {...props}
      >
        Value
      </FilterItem>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const event = {
      preventDefault: jest.fn(),
    };

    const component = shallow(
      <FilterItem
        {...props}
      >
        Value
      </FilterItem>,
    );

    component
      .find('a')
      .simulate('click', event);

    expect(props.onChange).toHaveBeenCalled();
    expect(event.preventDefault).toHaveBeenCalled();
  });
});
