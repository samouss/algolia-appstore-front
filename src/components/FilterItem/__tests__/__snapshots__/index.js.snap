// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterItem /> expect to render 1`] = `
<li>
  <a
    href=""
    onClick={[Function]}
    styleName="FilterItem"
  >
    Value
  </a>
</li>
`;

exports[`<FilterItem /> expect to render when refined 1`] = `
<li>
  <a
    href=""
    onClick={[Function]}
    styleName="FilterItem FilterItem--active"
  >
    Value
  </a>
</li>
`;
